plaidml/edsl/__init__.py

```diff
@@ -842,19 +842,31 @@ def floor(x):
     return intrinsic('floor', x)
 
 
-def gather(x, y):
+def gather(x, y, axis=0, interpolation_mode=1, nearest_mode=0, cube_coeff=-0.75):
     """Takes an input tensor (``x``) and a set of indices to gather over
     (``y``), and returns an output tensor that gathers the input tensor from the
     indices specified.
 
     Args:
         x (Tensor): The tensor to peform ``gather`` on.
         y (Tensor): The set of indices to ``gather`` over.
+        axis (int): The dimension index to gather data from.
+        interpolation_mode (int): The type of interpolation 
+            0: NEAREST
+            1: LINEAR
+            2: CUBIC
+        nearest_mode (int): The type of nearest interpolation 
+            0: ROUND_PREFER_FLOOR
+            1: ROUND_PREFER_CEIL
+            2: FLOOR,
+            3: CEIL,
+            4: SIMPLE
+        cube_coeff (float): The coefficient that controls the cubic interpolation.
 
     Returns:
         Tensor: The result of the ``gather`` operation.
     """
-    return intrinsic('gather', x, y)
+    return intrinsic('gather', x, y, axis, interpolation_mode, nearest_mode, cube_coeff)
```

Comment:
We should add enums to the python bindings as well.

Reply:
Added interpolation enums in this file.